terraform/cluster/bootstrap/templates/bootstrap_vm.tpl

```diff
@@ -32,9 +34,32 @@
 <source file='${bootstrap_img}'/>
 <backingStore/>
 <target dev='vda' bus='virtio'/>
+%{ if bootstrap_enable_boot_index == "true" ~}
+<boot order='1'/>
+%{ endif ~}
 <alias name='virtio-disk0'/>
 <address type='pci' domain='0x0000' bus='0x00' slot='0x06' function='0x0'/>
 </disk>
+%{ if bootstrap_enable_boot_index == "true" ~}
+<interface type='bridge'>
+<mac address='52:54:00:82:68:3e'/>
+<source bridge='${baremetal_bridge}'/>
+<target dev='vnet0'/>
+<model type='rtl8139'/>
+<alias name='net0'/>
+<boot order='3'/>
```

Comment:
In your PR comment, it says:

```
It is expected by etcd and kubeapi in 4.4 deployment that baremetal(API) interface in bootstrap VM is the default interface (with interface order being 1).
```

It looks like here, though, the baremetal interface is being set to a higher index (3) than the provisioning interface index (2).  Should it be lower than the provisioning index to make sure it's the default?

Reply:
There are two orders: one is boot order, configured via <boot order>, the other is interface order (when checking in OS with `ifconfig` or `ip link show`). provisioning interface needs to have lower boot order (hight boot priority) in order to provision the bootstrap VM from it.  baremetal interface needs to have lower interface order for etcd/kubeapi to get the right IP.

Reply:
interface order is simply determinzed by which interface appear as the first in bootstrap VM xml.

Reply:
Ah, okay.  I think PCI address also matters as well, which I see you've done too (baremetal has a lower PCI address than provisioning when `bootstrap_enable_boot_index` is true).

Reply:
right